File: 023_moudle/main.cpp
#include <iostream>

namespace my_module {

struct Point
{
    int x;
    int y;
};

void output(int x, int y)
{
    std::cout << x << " " << y << std::endl;
}

} // namespace my_module

int main(void)
{
    my_module::Point point = {10, 20};
    my_module::output(point.x, point.y);

    return 0;
}

File: 023_moudle/main.rs
mod my_module {
    pub struct Point {
        pub x: i32,
        pub y: i32,
    }

    pub fn output(x: i32, y: i32) {
        println!("{} {}", x, y);
    }
}

fn main() {
    let point = my_module::Point{x: 10, y: 20};
    my_module::output(point.x, point.y);
}

